packages/smooth_app/lib/pages/product/product_page/header/reorder_bottom_sheet.dart

```diff
@@ -0,0 +1,198 @@
+    this.onVisibilityToggle,
+    required this.title,
+  });
+
+  final List<ReorderableItem<T>> items;
+  final ValueChanged<List<ReorderableItem<T>>> onReorder;
+  final LabelBuilder<T> labelBuilder;
+  final ValueChanged<ReorderableItem<T>>? onVisibilityToggle;
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider<ReorderBottomSheetProvider<T>>(
+      create: (_) => ReorderBottomSheetProvider<T>(items),
+      child: Consumer<ReorderBottomSheetProvider<T>>(
+        builder:
+            (BuildContext context, ReorderBottomSheetProvider<T> provider, _) {
+          return DraggableScrollableSheet(
+            expand: false,
+            initialChildSize: 0.6,
+            maxChildSize: 0.9,
+            minChildSize: 0.4,
+            builder: (_, ScrollController scrollController) {
+              return ClipRRect(
+                borderRadius: const BorderRadius.vertical(top: ROUNDED_RADIUS),
+                child: Container(
+                  decoration: BoxDecoration(
+                    color: Theme.of(context).canvasColor,
+                    borderRadius: const BorderRadius.vertical(
+                      top: Radius.circular(16),
+                    ),
+                  ),
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      const SmoothModalSheetHeader(
+                        title: 'Reorder tabs',
+                      ),
+                      Expanded(
+                        child: ReorderableListView.builder(
+                          scrollController: scrollController,
+                          padding: const EdgeInsets.all(MEDIUM_SPACE),
+                          proxyDecorator: (
+                            Widget child,
+                            int index,
+                            Animation<double> animation,
+                          ) =>
+                              Transform.scale(
+                            scale: 1.0 + (0.05 * animation.value),
+                            child: Opacity(
+                              opacity: 0.8,
+                              child: child,
+                            ),
+                          ),
+                          itemBuilder: (BuildContext context, int index) {
+                            final ReorderableItem<T> item =
+                                provider.items[index];
+                            return Container(
+                              key: ValueKey<T>(item.data),
+                              margin:
+                                  const EdgeInsets.only(bottom: MEDIUM_SPACE),
+                              padding: const EdgeInsets.all(12.0),
+                              decoration: BoxDecoration(
+                                color: item.visible
+                                    ? SmoothColorsThemeExtension.defaultValues()
```

Comment:
Never use this. You force the creation of new objects.
Cf other parts of the code